diff --git a/dist/client/components/router-reducer/fetch-server-response.js b/dist/client/components/router-reducer/fetch-server-response.js
index db2687d458d4ee8af2f760e13357459429fb291c..0cc9560645a353fc5948398bdd9e8c7a2e1698f6 100644
--- a/dist/client/components/router-reducer/fetch-server-response.js
+++ b/dist/client/components/router-reducer/fetch-server-response.js
@@ -172,7 +172,7 @@ async function fetchServerResponse(url, options) {
         // TODO-APP: Add a test for the case where a CORS request fails, e.g. external url redirect coming from the response.
         // See https://github.com/vercel/next.js/issues/43605#issuecomment-1451617521 for a reproduction.
         return {
-            flightData: url.toString(),
+            flightData: urlToUrlWithoutFlightMarker(url.toString()).toString(),
             canonicalUrl: undefined,
             couldBeIntercepted: false,
             prerendered: false,
